firstvoices/backend/serializers/import_job_serializers.py

```diff
@@ -0,0 +1,143 @@
+
+from backend.models import Membership
+from backend.models.constants import Role
+from backend.models.import_jobs import ImportJob, ImportJobReport, ImportJobReportRow
+from backend.models.jobs import JobStatus
+from backend.models.media import File
+from backend.serializers import fields
+from backend.serializers.base_serializers import CreateSiteContentSerializerMixin
+from backend.serializers.job_serializers import BaseJobSerializer
+from backend.serializers.media_serializers import FileUploadSerializer
+from backend.serializers.utils.context_utils import get_site_from_context
+from backend.serializers.utils.import_job_utils import (
+    check_required_headers,
+    validate_headers,
+    validate_username,
+)
+from backend.serializers.validators import SupportedFileType
+
+
+class ImportReportRowSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ImportJobReportRow
+        fields = ["row_number", "status", "errors"]
+
+    row_number = serializers.IntegerField(read_only=True)
+    status = serializers.CharField(read_only=True)
+    errors = serializers.ListField(child=serializers.CharField(), read_only=True)
+
+
+class ImportReportSerializer(serializers.ModelSerializer):
+    total_rows = serializers.IntegerField(read_only=True)
+    diff_headers = serializers.ListField(child=serializers.CharField(), read_only=True)
+    rows = ImportReportRowSerializer(many=True, read_only=True)
+    totals = serializers.JSONField(read_only=True)
+
+    class Meta:
+        model = ImportJobReport
+        fields = ["total_rows", "diff_headers", "totals"]
+
+
+class ImportJobSerializer(CreateSiteContentSerializerMixin, BaseJobSerializer):
+    id = serializers.UUIDField(read_only=True)
+    data = FileUploadSerializer(
+        validators=[SupportedFileType(mimetypes=["text/csv", "text/plain"])],
+    )
+    run_as_user = serializers.CharField(required=False)
+
+    validation_task_id = serializers.CharField(read_only=True)
+    validation_status = fields.EnumField(enum=JobStatus, read_only=True)
+    validation_result = ImportReportSerializer(read_only=True)
+
+    class Meta:
+        model = ImportJob
+        fields = BaseJobSerializer.Meta.fields + (
+            "title",
+            "mode",
+            "validation_result",
+            "run_as_user",
+            "data",
+            "validation_task_id",
+            "validation_status",
+        )
+
+    def validate(self, attrs):
+        site = get_site_from_context(self)
+        user = self.context["request"].user
+        memberships = Membership.objects.filter(user=user)
+
+        run_as_user_input = attrs.get("run_as_user")
+        site_membership = memberships.filter(site=site).first()
+        if (
+            site_membership
+            and (site_membership.role in [Role.EDITOR, Role.LANGUAGE_ADMIN])
+            and run_as_user_input
+        ):
+            raise PermissionDenied(
+                "The runAsUser field can only be used by superadmins."
```

Comment:
Super minor: so far we don't say "superadmin" anywhere public. Let's just say "You don't have permission to use the runAsUser field."